src/plugin.js

```diff
@@ -109,18 +112,27 @@ const initPlugin = function(player, options) {
     }, 1000);
 
     player.clearTimeout(monitor);
+
     monitor = player.setTimeout(function() {
       // player already has an error
       // or is not playing under normal conditions
       if (player.error() || player.paused() || player.ended()) {
         return;
       }
 
+      // Disable timeouts in the background altogether according to the backgroundTimeout
+      // option, or if the player is muted, as browsers may throttle javascript timers to
+      // 1 minute in that case
+      if (document.visibilityState === 'hidden' && (options.backgroundTimeout === Infinity || player.muted())) {
```

Comment:
so, this only happens when the monitor goes off. Should we instead do this check inside `resetMonitor`? That way we don't even start the timer if we don't have to.

Reply:
Yeah, that thought occurred to me at the time but I opted to do it this way for some reason that I can no longer recall. Anyhow, it does make more sense the way you mentioned so I went ahead and made the change. I also reorganized some of the logic in `resetMonitor()` so its a bit easier to reason about.